### ISSUE TITLE
allow_js in ts_project does not contain js files in outputs for jsx inputs

### ISSUE THREAD

**Issue by noveyak** [bazelbuild/rules_nodejs#2390]
# 🐞 bug report

### Affected Rule

 The issue is caused by the rule:  ts_project

### Is this a regression?

I don't think so

### Description

When compiling .jsx files, tsc produces corresponding .js files but these are missing in the output of the rule

## 🔬 Minimal Reproduction

I'm hoping this is relatively easy to understand, I can create a small reproduction case but any .jsx input to ts_project with allow_js should reproduce

## 🔥 Exception or Error

No error and just no .js output

Looking at the ts_project rule, it seems like the outputs are created here

```
    if not emit_declaration_only:
        js_outs.extend(_out_paths(srcs, out_dir, root_dir, False, ".js"))
    if source_map and not emit_declaration_only:
        map_outs.extend(_out_paths(srcs, out_dir, root_dir, False, ".js.map"))
    if declaration or composite:
        typings_outs.extend(_out_paths(srcs, typings_out_dir, root_dir, allow_js, ".d.ts"))
    if declaration_map:
        typing_maps_outs.extend(_out_paths(srcs, typings_out_dir, root_dir, allow_js, ".d.ts.map"))

```

It seems like passing in `allow_js` instead of `False` to _out_paths for js_outs is enough to fix this problem but I'm not sure if there are other implications/undesirable effects of this change

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
ubuntu 16
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
bazel 3.7.0
  </code>
</pre>

**Rules_nodejs version:** 

(Please check that you have matching versions between WORKSPACE file and `@bazel/*` npm packages.)

<pre>
  <code>
3.0
  </code>
</pre>

**Anything else relevant?**

**Comment by alexeagle** [comment]
This would be a good issue to contribute if you'd like.

Add a test case under /packages/typescript/test/ts_project and make the code change you suggested, and see if everything else is green. `ts_project` has good coverage so if it's green it should be safe.

**Comment by mattem** [comment]
closed with #2398

**Comment by stefan-dimitrov** [comment]
The fix does not completely work - I cannot see the `.js` files in `bazel-out/`. It is there in the sandbox, but the jsx syntax is not converted - remains as in source (when using `jsx: "react"` in the tsconfig.json).

I'm running rules_nodejs version 4.4.5

Relevant tsconfig options:
```json
{
    "compilerOptions": {
        "target": "es5",
        "allowJs": true,
        "skipLibCheck": true,
        "composite": true,
        "lib": [
            "ES2015",
            "DOM"
        ],
        "jsx": "react",
        "esModuleInterop": true,
        "allowSyntheticDefaultImports": true,
        "strict": false,
        "forceConsistentCasingInFileNames": true,
        "module": "esnext",
        "moduleResolution": "Node",
    }
}
```
